### PR TITLE
Feature: content types + paramater encoding

### PR DESCRIPTION
**Pull request by vadymmarkov**
1) Add an enum for request method.
2) Add an enum for content type.
3) Introduce `ParameterEncoding` protocol for the case when you need to encode your request parameters to `NSData`.
4) Add default encoders for `JSON` and `FormURLEncoded` content types.
5) `WaveRider` is a suggested name for a basic configuration :surfer: 

screen shot 2016-02-26 at 01 18 26

### CODE REVIEW

File: Sources/ContentType.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    let string: String
+    
+    switch self {
+    case .JSON:
+      string = "application/json"
+    case .FormURLEncoded:
+      string = "application/x-www-form-urlencoded"
+    case .Custom(let value):
+      string = value
+    }
```
Comment (zenangst) [inline Sources/ContentType.swift]: I checked common content types but this seems to cover all our normal cases :sunglasses:

**Comment by zenangst** [comment]
This is a great start! :star2:

File: MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          let string = " \"#%<>[]\\^`{}|"
+          let result = "%20%22%23%25%3C%3E%5B%5D%5C%5E%60%7B%7D%7C"
+          
+          expect(encoder.escape(string)).to(equal(result))
+        }
+        
+        it("percent-escapes non-latin characters") {
+          expect(encoder.escape("Børk Børk Børk!!")).to(equal("B%C3%B8rk%20B%C3%B8rk%20B%C3%B8rk%21%21"))
+          expect(encoder.escape("українська мова")).to(equal("%D1%83%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D1%81%D1%8C%D0%BA%D0%B0%20%D0%BC%D0%BE%D0%B2%D0%B0"))
+          expect(encoder.escape("català")).to(equal("catal%C3%A0"))
```
Comment (vadymmarkov) [inline MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift]: This is for you @RamonGilabert

Comment (RamonGilabert) [inline MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift]: <3

File: MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        
+        it("percent-escapes illegal ASCII characters") {
+          let string = " \"#%<>[]\\^`{}|"
+          let result = "%20%22%23%25%3C%3E%5B%5D%5C%5E%60%7B%7D%7C"
+          
+          expect(encoder.escape(string)).to(equal(result))
+        }
+        
+        it("percent-escapes non-latin characters") {
+          expect(encoder.escape("Børk Børk Børk!!")).to(equal("B%C3%B8rk%20B%C3%B8rk%20B%C3%B8rk%21%21"))
```
Comment (zenangst) [inline MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift]: 🇸🇪🍲

File: Sources/Encoding/FormURLEncoder.swift [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+extension FormURLEncoder {
+  
+  func queryString(parameters: [String: AnyObject]) -> String {
+    var components: [Component] = []
+    
+    parameters.forEach { key, value in
+      components += queryComponents(key: key, value: value)
+    }
+    
+    return components.map({ "\($0)=\($1)" }).joinWithSeparator("&")
```
Comment (onmyway133) [inline Sources/Encoding/FormURLEncoder.swift]: Clean and concise

**Pull request merged**